src/app/forms/validators/custom-async-validators.ts

```diff
@@ -92,4 +93,28 @@ export class CustomAsyncValidators {
   static requiredIfNotAbandoned(store: Store<State>): AsyncValidatorFn {
     return this.requiredIfNotResult(store, resultOfTestEnum.abandoned);
   }
+  
+  static requiredIfNotResultAndSiblingEquals(store: Store<State>, result: any, sibling: string, value: any): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<ValidationErrors | null> =>
+      store.pipe(
+        take(1),
+        select(testResultInEdit),
+        map(testResult => {
+          if (control?.parent) {
+            const siblingControl = control.parent.get(sibling) as CustomFormControl;
+            const siblingValue = siblingControl.value;
+            const newValue = Array.isArray(value) ? value.includes(siblingValue) : siblingValue === value;
+
+            const currentResult = testResult?.testTypes[0].testResult;
+
+            if ((Array.isArray(result) ? !result.includes(currentResult) : currentResult !== result) && newValue && (control.value === null || control.value === undefined || control.value === '')) {
+              return { requiredIfNotResultAndSiblingEquals: { sibling: siblingControl.meta.label } };
+            }
+          }
+
+          return null;
+        })
+      );
+  }
+ 
```

Comment:
Why can you not attach the `requiredIfSiblingEquals` validator (or create it if it doesn't exist) and the `requiredIfResult` validators on the control?

Reply:
What I am trying to achieve is rules like: "particulateTrapFitted should only be required if modType code is p and the result is pass"

The problem is the validators are independent of each other so there is no way to do an "AND" condition as far as I know. If either validator fails then the error message is shown. Is there a way around that?